_includes/master/manifests/calico-config.yaml

```diff
@@ -0,0 +1,142 @@
+{% comment %}
+calico-config.yaml acccepts the following include flags:
+
+| Name         | Accepted Values          |
+|--------------|--------------------------|
+| datastore    | kdd, etcd                |
+| typha        | true, false              |
+| network      | calico, flannel, <unset> |
+| calico_ipam  | true, false              |
+| variant_name | Calico, Canal            |
+
+{% endcomment -%}
+# This ConfigMap is used to configure a self-hosted {{include.variant_name}} installation.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: {{include.variant_name | downcase}}-config
+  namespace: kube-system
+data:
+{%- if include.datastore == "etcd" %}
+  # Configure this with the location of your etcd cluster.
+  etcd_endpoints: "http://127.0.0.1:2379"
+
+  # If you're using TLS enabled etcd uncomment the following.
+  # You must also populate the Secret below with these files.
+  etcd_ca: ""   # "/calico-secrets/etcd-ca"
+  etcd_cert: "" # "/calico-secrets/etcd-cert"
+  etcd_key: ""  # "/calico-secrets/etcd-key"
+{%- elsif include.datastore == "kdd" and include.typha == "true" %}
+  # To enable Typha, set this to "calico-typha" *and* set a non-zero value for Typha replicas
+  # below.  We recommend using Typha if you have more than 50 nodes. Above 100 nodes it is
+  # essential.
+  typha_service_name: "none"
+{%- endif %}
+{%- if include.network == "calico" %}
+  # Configure the {{site.prodname}} backend to use.
+  calico_backend: "bird"
```

Comment:
not super important, but this is smushing with typha_service_name in the kdd manifest:

```
data:
  # To enable Typha, set this to "calico-typha" *and* set a n
  # below.  We recommend using Typha if you have more than 50
  # essential.
  typha_service_name: "none"
  # Configure the Calico backend to use.
  calico_backend: "bird"
```

Reply:
never mind, forget i mentioned this. there's a few places i'm seeing this, but we can fix it later.